packages/react/dialog/src/Dialog.tsx

```diff
@@ -82,7 +92,7 @@ const DialogTrigger = React.forwardRef((props, forwardedRef) => {
       {...triggerProps}
       as={as}
       ref={composedTriggerRef}
-      onClick={composeEventHandlers(props.onClick, () => context.onOpenChange(true))}
+      onClick={composeEventHandlers(props.onClick, context.onOpenToggle)}
```

Comment:
Why do dialog triggers toggle now? Is that a thing? 🤔

Reply:
Now that this introduces non-modal dialogs then yes you can click the trigger again, so feels like we should have the same default as for popover for example.

Reply:
This still feels weird to me but i guess we can just go with it for now and see what feedback we get. Unless @colmtuite has any input?